src/ui/box/header.jsx

```diff
@@ -6,6 +6,13 @@ import { BackButton } from './button';
 // TODO: simplify this mess :)
 
 export default class Header extends React.Component {
+  setHeaderHeightStyle(headerElement) {
+    if (headerElement) {
+      const height = headerElement.clientHeight || 0;
+      document.documentElement.style.setProperty('--header-height', `${height}px`);
+    }
+  }
```

Comment:
Would appreciate some ideas on how to test this. I would like to somehow mock the getting of the ref but I can't seem to do this properly with `jest.spyOn(..)`.

References: 

[How to mock ref properties](https://github.com/enzymejs/enzyme/issues/1937) (GH)
[How to mock ref properties](https://stackoverflow.com/questions/53721999/react-jest-enzyme-how-to-mock-ref-properties) (SO)

Reply:
I put a suggestion here: https://github.com/auth0/lock/compare/fix/header-height...header-height-test

Reply:
Thanks. Unfortunately it suffers from the same problem that the current master has - because it's executed from `componentWillMount`, the height is set once but then never recalculated if the height changes (switching tabs, etc).

However, I've managed to fix this using `createNodeMock`: https://github.com/facebook/react/issues/7740#issuecomment-247335106

Added a new test in https://github.com/auth0/lock/pull/1859/commits/e8c0dbf81c3327a73ff8259b034dc04e1948fa72